### DISCUSSION TITLE
WASM Project Build Failin on Mac OS

### DISCUSSION THREAD

**Discussion by cig-max** [unoplatform/uno#16647] [General]
Hi,

I created new project with this command that was generated by Uno Platform Template Wizard: `dotnet new unoapp -o MyApp -preset "recommended" -auth "custom" -server  -theme-service  -ci "github" -maui  -wasm-multi-threading  -theme "material" -presentation "mvux" -config  -di  -log "serilog" -nav "regions" -http  -loc  -tests "ui" -tests "unit" -toolkit  -dsp  -id "com.mycompany.myapp" -pub "O=MyCompany"`.

I am using Rider IDE (2024.1.1) on my Mac Book Pro (Intel CPU) and getting this compilation error:
```
3>Uno.Wasm.Bootstrap.targets(213,3): Error  : System.Exception: Failed to run emscripten. More details are available in diagnostics logging mode or using the MSBuild /bl switch, see https://aka.platform.uno/msbuild-troubleshoot.
   at Uno.Wasm.Bootstrap.ShellTask_v22722a164cfb242bfbdc2b10248f624fa32b28c2.RunPackager() in /_/src/Uno.Wasm.Bootstrap/ShellTask.cs:line 968
   at Uno.Wasm.Bootstrap.ShellTask_v22722a164cfb242bfbdc2b10248f624fa32b28c2.Execute() in /_/src/Uno.Wasm.Bootstrap/ShellTask.cs:line 270
```

I ran Uno Platform Check and did not find any issues (Uno Platform Check v1.22.1.0) with local setup.
I also tried to build same solution on another Mac Book Pro (Silicon CPU) with same version on Uno Platform SDK and same version of Rider IDE and got same error.

Thanks,
Max.

**Comment by jeromelaban** [comment]
Thanks for the report. To further troubleshoot this issue, you'll need to open the binlog with the steps at the url mentioned in the error message: https://aka.platform.uno/msbuild-troubleshoot.
